airbyte-integrations/connectors/source-mailgun/source_mailgun/source.py

```diff
@@ -0,0 +1,154 @@
+
+
+# Basic full refresh stream
+class MailgunStream(HttpStream, ABC):
+    # TODO: Support regions (EU and US) for domains
+    url_base = "https://api.mailgun.net/v3/"
+
+    primary_key = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, HttpUrl]]:
+        """
+        :param response: the most recent response from the API
+        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        next_page: Optional[HttpUrl] = response.json().get('paging', {}).get('next')
+        return {"url": next_page} if next_page and self._pre_parse_response(response) else None
+
+    def path(
+            self,
+            stream_state: Mapping[str, Any] = None,
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Optional[Mapping[str, HttpUrl]] = None,
+    ) -> str:
+        # TODO: Requires improve URL creation in the CDK by using urllib.parse.urljoin
+        #  (airbyte_cdk.sources.streams.http.http._create_prepared_request)
+        return next_page_token["url"] if next_page_token else ""
+
+    def request_params(
+            self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None,
+            next_page_token: Optional[Mapping[str, Any]] = None
+    ) -> MutableMapping[str, Any]:
+        return {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        yield from self._pre_parse_response(response)
+
+    @staticmethod
+    def _pre_parse_response(response: requests.Response) -> List:
+        return response.json()['items']
+
+
+class Domains(MailgunStream):
+    primary_key = "name"
+
+    def path(self, *args, next_page_token: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        # return super().path(*args, **kwargs) or "domains"  # TODO: Requires the CDK update (see MailgunStream.path())
+        path: str = "domains"
+        if next_page_token:
+            token: str = next_page_token['url'].rpartition('/')[2]
+            path = f"{path}/{token}"
+
+        return path
+
+
+# Basic incremental stream
+class IncrementalMailgunStream(MailgunStream, ABC):
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> \
+            Mapping[str, Any]:
+        return {}
+
+
+class Events(IncrementalMailgunStream):
+    # TODO: Event Polling. See https://documentation.mailgun.com/en/latest/api-events.html#event-polling
+
+    cursor_field = "timestamp"
+
+    primary_key = "id"
+
+    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.timestamp = config.get('timestamp', 0)
```

Comment:
I think names like start_timestamp, start_ts, start_datetime reflex better the purpose of this attribute.

Reply:
Also, it is a best practice to include start in connector specification to set minimal cursor value for all incrementall streams.